### ISSUE TITLE
[DISC] How to specify ports for authentication 

### ISSUE THREAD

**Issue by hekaldama** [mesos/mesos-go#229]
I have figured out how to specify a port for my scheduler via `DriverConfig.BindingPort` but can't seem to figure out how to do this for the authentication step. I get output like:

```
I0304 14:57:51.034809   20265 scheduler.go:839] authenticating with master master@10.1.79.45:5050
I0304 14:57:51.034985   20265 http_transporter.go:407] listening on 10.67.12.72 port 36486
```

and I would like to be able to set that port to a known value. My end goal is to be able to run a framework inside of a docker container without having to use `--net host`.

Let me know if I can provide anymore information or if that doesn't make sense.

Thank you.

**Comment by jdef** [comment]
I don't think we got around to implementing that. The original goal was to
be able to share the same underlying Transport/Messenger so that an
additional port wouldn't be necessary, but we never got that far.

The new version of the bindings (the "next" branch) that use the Mesos HTTP
v1 API won't have this problem because it does away with network PIDs.
Those APIs are still experimental though and "next" is under development.

In the near term, perhaps we should consider a less disruptive change (vs.
sharing as per above) and instead add a property for the auth context that
allows one to specify the authn binding port at framework boostrap time. It
doesn't feel like a particularly elegant solution, but it would probably
get the job done. One potential issue is that of socket reuse in Go: it's
not possible to specify REUSE_ADDR, so if registration starts flapping at
all the authn component could have temporary problems binding to the same
port if the previous one is stuck in TIME_WAIT

**Comment by jdef** [comment]
resolved by #252
